applications/asset_tracker_v2/src/modules/modem_module.c

```diff
@@ -1075,6 +1080,65 @@ static void on_all_states(struct modem_msg_data *msg)
 	}
 }
 
+void pdn_event_handler(uint8_t cid, enum pdn_event event, int reason)
+{
+	switch (event) {
+	case PDN_EVENT_CNEC_ESM:
+		LOG_WRN("Event: PDP context %d, %s", cid, esm_strerr(reason));
+		break;
+	case PDN_EVENT_ACTIVATED:
+		LOG_WRN("PDN_EVENT_ACTIVATED");
+		{ SEND_EVENT(modem, MODEM_EVT_LTE_CONNECTED); }
+		break;
+	case PDN_EVENT_DEACTIVATED:
+		LOG_WRN("PDN_EVENT_DEACTIVATED");
+		{ SEND_EVENT(modem, MODEM_EVT_LTE_DISCONNECTED); }
+		break;
+	case PDN_EVENT_IPV6_UP:
+		LOG_WRN("PDN_EVENT_IPV6_UP");
+		break;
+	case PDN_EVENT_IPV6_DOWN:
+		LOG_WRN("PDN_EVENT_IPV6_DOWN");
+		break;
+	default:
+		printk("Event: PDP context %d %s\n", cid, event_str[event]);
+		break;
+	}
+}
+
+static int pdn_setup(void)
+{
+	int err;
+
+	/* Register to the necessary packet domain AT notifications. */
+	err = nrf_modem_at_printf("AT+CNEC=16");
+	if (err) {
+		LOG_ERR("AT+CNEC=16 failed, err %d", err);
+		return err;
+	}
+
+	err = nrf_modem_at_printf("AT+CGEREP=1");
+	if (err) {
+		LOG_ERR("AT+CGEREP=1 failed, err %d", err);
+		return err;
+	}
+
+	err = pdn_init();
+	if (err) {
+		LOG_ERR("pdn_init failed, error: %d", err);
+		return err;
+	}
```

Comment:
Once https://github.com/nrfconnect/sdk-nrf/pull/7303 is merged, the library will subscribe to these automagically. `pdn_init()` is also removed.